components/proxy/src/test/kotlin/com/hotels/styx/admin/handlers/RoutingObjectHandlerTest.kt

```diff
@@ -0,0 +1,274 @@
+                            .body("""
+                            name: "staticResponse"
+                            type: "StaticResponseHandler"
+                            tags: []
+                            config:
+                               status: 200
+                               content: "Hello, world!"
+                            """.trimIndent(),
+                                    UTF_8)
+                            .build()
+                            .stream(), HttpInterceptorContext.create())
+                    .flatMap { it.aggregate(2000) }
+                    .toMono()
+                    .block()
+                    ?.status() shouldBe CREATED
+
+            routeDatabase.get("staticResponse").isPresent shouldBe true
+            routeDatabase.get("staticResponse").get().type shouldBe "StaticResponseHandler"
+            routeDatabase.get("staticResponse").get().config.shouldBeTypeOf<ObjectNode>()
+            routeDatabase.get("staticResponse").get().handler.shouldBeTypeOf<StaticResponseHandler>()
+        }
+
+        scenario("Retrieving objects") {
+            val handler = RoutingObjectHandler(routeDatabase, objectFactory)
+
+            handler.handle(
+                    HttpRequest.put("/admin/routing/objects/staticResponse")
+                            .body("""
+                            type: StaticResponseHandler
+                            config:
+                               status: 200
+                               content: "Hello, world!"
+                            """.trimIndent(),
+                                    UTF_8)
+                            .build()
+                            .stream(), HttpInterceptorContext.create())
+                    .flatMap { it.aggregate(2000) }
+                    .toMono()
+                    .block()
+                    ?.status() shouldBe CREATED
+
+            val response = handler.handle(HttpRequest.get("/admin/routing/objects/staticResponse").build().stream(), HttpInterceptorContext.create())
+                    .flatMap { it.aggregate(2000) }
+                    .toMono()
+                    .block()
+
+            response!!.status() shouldBe OK
+            response.bodyAs(UTF_8).trim() shouldBe """
+                            ---
+                            name: "staticResponse"
+                            type: "StaticResponseHandler"
+                            tags: []
+                            config:
+                              status: 200
+                              content: "Hello, world!"
+                              """
+                    .trimIndent()
+                    .trim()
+        }
+
+        scenario("Fetching all routing objects") {
+            val handler = RoutingObjectHandler(routeDatabase, objectFactory)
+
+            handler.handle(
+                    HttpRequest.put("/admin/routing/objects/staticResponse")
```

Comment:
We should be able to import these `HttpRequest` factory methods (`put`, `get` etc.)